aws/resource_aws_appconfig_configuration_profile.go

```diff
@@ -0,0 +1,301 @@
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(0, 1024),
+				),
+			},
+			"location_uri": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 2048),
+				),
+			},
+			"retrieval_role_arn": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateArn,
+			},
+			"validator": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{appconfig.ValidatorTypeJsonSchema, appconfig.ValidatorTypeLambda}, false),
+						},
+						"content": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"tags": tagsSchema(),
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsAppconfigConfigurationProfileCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appconfigconn
+	applicationID := aws.String(d.Get("application_id").(string))
+	configProfileName := aws.String(d.Get("name").(string))
+	configProfileDescription := aws.String(d.Get("description").(string))
+	locationURI := aws.String(d.Get("location_uri").(string))
+	retreivalRoleArn := aws.String(d.Get("retrieval_role_arn").(string))
+	if *retreivalRoleArn == "" {
+		retreivalRoleArn = nil
+	}
+
+	var validatorList []*appconfig.Validator
+	if validators := d.Get("validator").(*schema.Set).List(); len(validators) > 0 {
+		validatorList = convertMapToValidators(validators)
+	}
+
+	input := &appconfig.CreateConfigurationProfileInput{
+		ApplicationId:    applicationID,
+		Name:             configProfileName,
+		Description:      configProfileDescription,
+		LocationUri:      locationURI,
+		RetrievalRoleArn: retreivalRoleArn,
+		Validators:       validatorList,
+		Tags:             keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().AppconfigTags(),
+	}
+
+	expectedErrMsg := fmt.Sprintf("Error trying to assume role %s", d.Get("retrieval_role_arn").(string))
+	expectedErrMsg2 := "is not authorized to perform the operation"
+	var config *appconfig.CreateConfigurationProfileOutput
+	var err error
+	err = resource.Retry(iamwaiter.PropagationTimeout, func() *resource.RetryError {
+		config, err = conn.CreateConfigurationProfile(input)
+		if err != nil {
+			if isAWSErr(err, "BadRequestException", expectedErrMsg) || isAWSErr(err, "BadRequestException", expectedErrMsg2) {
+				log.Printf("[DEBUG] Retrying AppConfig Configuration Profile creation: %s", err)
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+	if isResourceTimeoutError(err) {
```

Comment:
is this doing another round of retries outside of the resource.Retry magic? 
Why is that?

Reply:
Yeah, that's what they do in [the eventual consistency doc](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/retries-and-waiters.md#eventual-consistency). They say that block "Retry AWS Go SDK operation if no response from automatic retries."